src/components/Monaco.vue

```diff
@@ -3,100 +3,108 @@
 </template>
 
 <script>
-import * as monaco from 'monaco-editor'
-import { computed, onMounted, onUnmounted, ref, watch } from 'vue'
-// import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
-// import { createMonacoComplete, createMonacoHover } from "@/common/monacoUtils";
+import { defineComponent } from 'vue'
 
-export default {
+export default defineComponent({
   name: 'Monaco',
+})
+</script>
 
-  props: {
-    id: {
-      type: String,
-      required: true,
-    },
-    value: {
-      type: String,
-      required: true,
-    },
-    lang: {
-      type: String,
-      required: true,
-    },
-    disabled: {
-      type: Boolean,
-      default: false,
-    },
-    // warp: {
-    //   type: Boolean,
-    //   default: false,
-    // },
-    // provider: {
-    //   type: Array,
-    //   default: () => [],
-    // },
-    scrollLoading: {
-      type: Boolean,
-      default: false,
-    },
-    scrollFunc: {
-      type: Function,
-      default: () => () => {},
-    },
+<script setup>
+import * as monaco from 'monaco-editor'
+import { defineProps, defineEmits, onMounted, onUnmounted, watch } from 'vue'
+
+const prop = defineProps({
+  id: {
+    type: String,
+    required: true,
   },
-  setup(prop) {
-    const editor = ref({})
-    // const editValue = computed(() => {
-    //   console.log(prop.value + "changed");
-    //   return prop.value;
-    // });
+  modelValue: {
+    type: String,
+    required: true,
+  },
+  lang: {
+    type: String,
+    required: true,
+  },
+  disabled: {
+    type: Boolean,
+    default: false,
+  },
+  // warp: {
+  //   type: Boolean,
+  //   default: false,
+  // },
+  // provider: {
+  //   type: Array,
+  //   default: () => [],
+  // },
+  scrollLoading: {
+    type: Boolean,
+    default: false,
+  },
+  scrollFunc: {
+    type: Function,
+    default: () => () => {},
+  },
+})
 
-    const initEditor = () => {
-      const id = `monaco-${prop.id}`
-      const defaultOptions = {
-        value: prop.value,
-        language: prop.lang,
-        readOnly: prop.disabled,
-        // fontSize: 12,
-        automaticLayout: true,
-        scrollBeyondLastLine: false,
-        // theme: "vs",
-        minimap: {
-          enabled: false,
-        },
-        scrollbar: {
-          verticalScrollbarSize: 6,
-          horizontalScrollbarSize: 6,
-        },
-        // hover: {
-        //   delay: 500,
-        //   enabled: true,
-        // },
-      }
+const emit = defineEmits(['update:modelValue'])
 
-      editor.value = monaco.editor.create(document.getElementById(id), defaultOptions)
+// ❗️ editor instance can not be reactive, otherwise it will cause the page to get stuck for unknown reasons
+let editor = {}
+// const editValue = computed(() => {
+//   console.log(prop.modelValue + "changed");
+//   return prop.modelValue;
+// });
```

Comment:
OK, It shouldn't matter if you don't use reactive here, what does the pseudo-code comment mean?

Reply:
It's the previous code, I always don't want to delete the previously commented code, what if I need it later?

Reply:
It looks like for debugging... Ok, just remove it?

Reply:
BTW, I think the init data of the `editor` will be `null`?